vms/platformvm/txs/executor/standard_tx_executor_test.go

```diff
@@ -1110,205 +1091,122 @@ func TestDurangoMemoField(t *testing.T) {
-				chainTime := env.state.GetTimestamp()
-				createBlockchainTxFee := env.config.GetCreateBlockchainTxFee(chainTime)
-
-				ins, unstakedOuts, _, signers, err := env.utxosHandler.Spend(
-					env.state,
+			setupTest: func(env *environment, memoField []byte) (*txs.Tx, state.Diff) {
+				tx, err := env.txBuilder.NewCreateChainTx(
+					testSubnet1.TxID,
+					[]byte{},             // genesisData
+					ids.GenerateTestID(), // vmID
+					[]ids.ID{},           // fxIDs
+					"aaa",                // chain name
 					preFundedKeys,
-					defaultMinValidatorStake,
-					createBlockchainTxFee,
 					ids.ShortEmpty,
+					memoField,
 				)
 				require.NoError(t, err)
 
-				subnetAuth, subnetSigners, err := env.utxosHandler.Authorize(env.state, testSubnet1.TxID, preFundedKeys)
-				require.NoError(t, err)
-				signers = append(signers, subnetSigners)
-
 				onAcceptState, err := state.NewDiff(env.state.GetLastAccepted(), env)
 				require.NoError(t, err)
 
-				tx := &txs.CreateChainTx{
-					BaseTx: txs.BaseTx{BaseTx: avax.BaseTx{
-						NetworkID:    env.ctx.NetworkID,
-						BlockchainID: env.ctx.ChainID,
-						Ins:          ins,
-						Outs:         unstakedOuts,
-					}},
-					SubnetID:    testSubnet1.TxID,
-					ChainName:   "aaa",
-					VMID:        ids.GenerateTestID(),
-					FxIDs:       []ids.ID{},
-					GenesisData: []byte{},
-					SubnetAuth:  subnetAuth,
-				}
-				return tx, signers, onAcceptState, &tx.Memo
+				return tx, onAcceptState
 			},
 		},
 		{
 			name: "CreateSubnetTx",
-			setupTest: func(env *environment) (txs.UnsignedTx, [][]*secp256k1.PrivateKey, state.Diff, *types.JSONByteSlice) {
-				chainTime := env.state.GetTimestamp()
-				createSubnetTxFee := env.config.GetCreateSubnetTxFee(chainTime)
-
-				ins, unstakedOuts, _, signers, err := env.utxosHandler.Spend(
-					env.state,
+			setupTest: func(env *environment, memoField []byte) (*txs.Tx, state.Diff) {
+				tx, err := env.txBuilder.NewCreateSubnetTx(
+					1,
+					[]ids.ShortID{ids.GenerateTestShortID()},
 					preFundedKeys,
-					defaultMinValidatorStake,
-					createSubnetTxFee,
 					ids.ShortEmpty,
+					memoField,
 				)
 				require.NoError(t, err)
 
 				onAcceptState, err := state.NewDiff(env.state.GetLastAccepted(), env)
 				require.NoError(t, err)
 
-				tx := &txs.CreateSubnetTx{
-					BaseTx: txs.BaseTx{BaseTx: avax.BaseTx{
-						NetworkID:    env.ctx.NetworkID,
-						BlockchainID: env.ctx.ChainID,
-						Ins:          ins,
-						Outs:         unstakedOuts,
-					}},
-					Owner: &secp256k1fx.OutputOwners{
-						Threshold: 1,
-						Addrs:     []ids.ShortID{ids.ShortEmpty},
-					},
-				}
-				return tx, signers, onAcceptState, &tx.Memo
+				return tx, onAcceptState
 			},
 		},
 		{
 			name: "ImportTx",
-			setupTest: func(env *environment) (txs.UnsignedTx, [][]*secp256k1.PrivateKey, state.Diff, *types.JSONByteSlice) {
+			setupTest: func(env *environment, memoField []byte) (*txs.Tx, state.Diff) {
```

Comment:
This test is really the reason why I added memo fields in the txBuilder functions.
With the activation of Durango, we zeroed memo fields, so it may seem weird that I introduce them back.
However txBuilder is going to become a test package and we may want to be able to setup memos in tests.

Reply:
Currently memo fields are disabled, but we may consider reintroducing them once we properly price txs.
I think generally we want to allow maximal flexibility in building txs, especially in UTs.
Note for instance that with Durango activation we won't need to specify StartTime in any of the AddValidator/Delegator txs but we still explicitly pass start time in the builder signature.
Not sure I understand what should go in the README here?